Tests/CatsAppTests/NetworkClientTests.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  NetworkClientTests.swift
+//  CatsAppTests
+//
+//  Created by Carlos Costa on 08/10/2025.
+//
+
+import Foundation
+import Testing
+
+@testable import CatsApp
+
+class MockClient: NetworkClientProtocol {
+    var resultData: Data?
+    var statusCode: Int = 200
+    var error: Error?
+
+    func request<T: Decodable>(_ endpoint: Endpoint) async throws -> T where T: Decodable {
+        if let error = error {
+            throw error
+        }
+        if !(200...299).contains(statusCode) {
+            throw NetworkError.serverStatus(statusCode)
+        }
+        guard let data = resultData else {
+            throw NetworkError.unknown
+        }
+        do {
+            let decoded = try JSONDecoder().decode(T.self, from: data)
+            return decoded
+        } catch {
+            throw NetworkError.decoding(error)
+        }
+    }
+}
+
+@Suite("NetworkClient Success Test")
+struct NetworkClientTests {
+    @Test("Successful request decodes MockDTO")
+    func testSuccess() async throws {
+        let client = MockClient()
+        let json = try JSONEncoder().encode(MockDTO.breedsDTO)
+        client.resultData = json
+        let endpoint = Endpoint(path: "breeds")
+        let breeds: [BreedsDataService.CatBreed] = try await client.request(endpoint)
+        #expect(!breeds.isEmpty, "Breeds should not be empty")
+        #expect(breeds.count == MockDTO.breedsDTO.count, "Breeds count should match mock data")
+        #expect(breeds.first?.id == "abys", "First breed should be Abyssinian")
+    }
+}
+
+@Suite("NetworkClient Failure Test")
+struct NetworkClientErrorTests {
+    
+    @Test("Throws decoding error for empty data")
+    func testEmptyDataDecodingError() async {
+        let client = MockClient()
+        client.resultData = Data()
+        let endpoint = Endpoint(path: "breeds")
+        do {
+            _ = try await client.request(endpoint) as [BreedsDataService.CatBreed]
+            #expect(Bool(false), "Expected decoding error for empty data, but got success")
+        } catch {
+            if let netErr = error as? NetworkError {
+                if case .decoding = netErr {
+                    #expect(true)
+                } else {
+                    #expect(Bool(false), "Expected decoding error, got \(netErr)")
+                }
+            } else {
+                #expect(Bool(false), "Expected NetworkError but got \(type(of: error))")
+            }
+        }
+    }
```

Comment:
- Not a 100% sure this would work/compile but I think you can simplify this a bit with the `try await #expect(throws:)`. 
- We can also remove the `#expect(true)` since if we do not hit the `#expect(false)`, the test will succeed.
- No need to convert `false` into a `Bool`, it's already a boolean 😄 -> `#expect(false, ...)`

You can try and apply these suggestions for the other tests as well 

```suggestion
func testEmptyDataDecodingError() async throws { // Add a 'throws' here
    let client = MockClient()
    client.resultData = Data()
    let endpoint = Endpoint(path: "breeds")

    let thrownError = try await #expect(throws: client.request(endpoint) as [BreedsDataService.CatBreed])

    guard case .decoding = thrownError as? NetworkError else {
        #expect(false, "Expected NetworkError.decoding, but got \(thrownError)")
        return
    }
}
```

Reply:
Damn, my solution compared to yours is really verbose 🥲 That definitely makes it cleaner. I’ll update the tests and simplify the rest too 👍 Thank you!

Reply:
I tried the approach with `#expect(throws:)`like you suggested, but unfortunately it doesn’t compile 🥲
So I refactored the tests using the original `do/catch` style, but cleaned them up a bit.